internal/providers/providers.go

```diff
@@ -242,6 +245,51 @@ func (pb *ProviderBuilder) GetRepoLister() (provinfv1.RepoLister, error) {
 	return nil, fmt.Errorf("provider does not implement repo lister")
 }
 
+// GetGHCR returns a GHCR client for the provider.
+func (pb *ProviderBuilder) GetGHCR() (provinfv1.GHCR, error) {
+	if !pb.Implements(db.ProviderTypeGhcr) {
```

Comment:
If we had the ghcr provider as part of the github interface we could do:
```
if pb.Implements(db.ProviderTypeGithub) { return pb.GetGithub }
```

Reply:
right, that could be the case; but the intent was to be able to have an independent ghcr provider.